common/gcp_test.go

```diff
@@ -26,6 +41,69 @@ func TestGoogleOIDC(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestConfigureWithGoogleCredentialsUsingInvalidJson(t *testing.T) {
+	defer CleanupEnvironment()()
+	client := &DatabricksClient{
+		Host:                 "https://accounts.gcp.databricks.com/",
+		GoogleServiceAccount: "my-sa2@fake-project-id.iam.gserviceaccount.com",
+		GoogleCredentials:    fakeCredentialsJson,
+	}
+	client.configureHTTPCLient()
+
+	_, err := client.configureWithGoogleCrendentials(context.Background())
+	assert.ErrorContains(t, err,
+		"could not obtain OIDC token from JSON: oauth2: cannot fetch token")
+	// TODO: To get better test coverage, start a local OAuth2 server to support
```

Comment:
We're doing this all over the place :) see azure auth tests, for example.

Reply:
https://github.com/databrickslabs/terraform-provider-databricks/blob/master/common/azure_auth_test.go#L143-L203 ain't pretty, but they do work.